sklearn/linear_model/ridge.py

```diff
@@ -1136,52 +1142,75 @@ def fit(self, X, y, sample_weight=None):
                 out, c = _errors(float(alpha), y, v, Q, QT_y)
             else:
                 out, c = _values(float(alpha), y, v, Q, QT_y)
-            cv_values[:, i] = out.ravel()
+            cv_values[:, :, i] = out
             C.append(c)
 
+        if self.store_cv_values:
+            self.cv_values_ = cv_values
+
         if error:
-            best = cv_values.mean(axis=0).argmin()
+            if self.alpha_per_target:
+                # Find the best alpha for each target
+                best = cv_values.mean(axis=0).argmin(axis=1)
+            else:
+                # Find the best alpha overall
+                best = np.mean(cv_values.reshape(-1, len(self.alphas)),
+                               axis=0).argmin()
         else:
-            # The scorer want an object that will make the predictions but
+            # The scorer wants an object that will make the predictions, but
             # they are already computed efficiently by _RidgeGCV. This
             # identity_estimator will just return them
             def identity_estimator():
                 pass
             identity_estimator.decision_function = lambda y_predict: y_predict
             identity_estimator.predict = lambda y_predict: y_predict
 
-            out = [scorer(identity_estimator, y.ravel(), cv_values[:, i])
-                   for i in range(len(self.alphas))]
-            best = np.argmax(out)
+            if self.alpha_per_target:
+                out = [scorer(identity_estimator, target, cv_values[:, i, j])
+                       for j in range(len(self.alphas))
+                       for i, target in enumerate(y.T)]
+                best = np.argmax(out, axis=1)
+            else:
+                out = [scorer(identity_estimator, y.ravel(),
+                              cv_values[:, :, i].ravel())
+                       for i in range(len(self.alphas))]
+                best = np.argmax(out)
 
         self.alpha_ = self.alphas[best]
-        self.dual_coef_ = C[best]
+
+        if self.alpha_per_target:
+            self.dual_coef_ = np.vstack([C[j][:, i]
```

Comment:
We should test `dual_coef_`

Reply:
I think we already do:

https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/tests/test_ridge.py#L100

Reply:
As in the shape of `dual_coef_` is what we expected.